pkg/autoscaler/config/config.go

```diff
@@ -248,6 +278,12 @@ func validate(lc *Config) (*Config, error) {
 		return nil, fmt.Errorf("panic-window-percentage = %v, must be in [%v, 100] interval", lc.PanicWindowPercentage, 100*float64(BucketSize)/float64(lc.StableWindow))
 	}
 
+	if lc.InitialScale < 0 {
+		return nil, fmt.Errorf("initial-scale = %v, must be at least 0 (or at least 1 when allow-zero-initial-scale is false)", lc.InitialScale)
+	}
+	if lc.InitialScale == 0 && !lc.AllowZeroInitialScale {
+		return nil, fmt.Errorf("initial-scale = %v, must be at least 1 since allow-zero-initial-scale is false", lc.InitialScale)
+	}
```

Comment:
If we re-arrange in the reverse order, then we can go back to the previous errors messages, since they'd be correct, I think. But probably isn't important much.

Reply:
I'm not sure I understand. What do you mean by "we can go back to the previous errors messages"?

Reply:
eh, basically
```
if x < 0 || x==0 && !allow {
  return "must be non-negative or positive if allow is unser"
}
```

Reply:
How about:
```go
minInitialScale := 1
if lc.AllowZeroInitialScale {
  minInitialScale = 0
}
if lc.InitialScale < minInitialScale {
  return nil, fmt.Errorf("initial-scale = %d, must be at least %d", lc.InitialScale, minInitialScale)
}
```

Reply:
Well we kind of had that before :-) 
It's bikeshedding at this point. I like mine, since it has single `if` :)